src/platform/packages/shared/kbn-storage-adapter/src/index_adapter/index.ts

```diff
@@ -375,28 +330,15 @@ export class StorageIndexAdapter<
     ...request
   }): Promise<StorageClientIndexResponse> => {
     const attemptIndex = async (): Promise<IndexResponse> => {
-      const [danglingItem] = id ? await this.getDanglingItems({ ids: [id] }) : [undefined];
-
-      const [indexResponse] = await Promise.all([
```

Comment:
This implementation can lead to data loss if the `.index` request fails but the `.delete` succeeds (or data duplication in the reverse scenario). Supporting OLTP data in this way can also result in sparsely populated indices over time.

If we agree on the single index vision then we remove the dangling data clean up logic for now.